applications/StructuralMechanicsApplication/tests/cpp_tests/test_tangent_tensor_numerical_derivation_utility.cpp

```diff
@@ -16,106 +17,136 @@
+ * @brief This test tests that the perturbation utility is valid for computing the elastic linear tensor
+ */
+KRATOS_TEST_CASE_IN_SUITE(LinearElasticCasePertubationTensorUtility, KratosStructuralMechanicsFastSuite)
 {
-    const double E = rMaterialProperties[YOUNG_MODULUS];
-    const double poisson_ratio = rMaterialProperties[POISSON_RATIO];
-    const double lambda =
-        E * poisson_ratio / ((1. + poisson_ratio) * (1.0 - 2.0 * poisson_ratio));
-    const double mu = E / (2.0 + 2.0 * poisson_ratio);
-
-    if (rElasticityTensor.size1() != 6 || rElasticityTensor.size2() != 6)
-        rElasticityTensor.resize(6, 6, false);
-    rElasticityTensor.clear();
-
-    rElasticityTensor(0, 0) = lambda + 2.0 * mu;
-    rElasticityTensor(0, 1) = lambda;
-    rElasticityTensor(0, 2) = lambda;
-    rElasticityTensor(1, 0) = lambda;
-    rElasticityTensor(1, 1) = lambda + 2.0 * mu;
-    rElasticityTensor(1, 2) = lambda;
-    rElasticityTensor(2, 0) = lambda;
-    rElasticityTensor(2, 1) = lambda;
-    rElasticityTensor(2, 2) = lambda + 2.0 * mu;
-    rElasticityTensor(3, 3) = mu;
-    rElasticityTensor(4, 4) = mu;
-    rElasticityTensor(5, 5) = mu;
+    ConstitutiveLaw::Parameters cl_configuration_values;
+    Properties material_properties;
+    Vector stress_vector, strain_vector;
+    ProcessInfo CurrentProcessInfo;
+
+    Flags constitutive_law_options;
+    constitutive_law_options.Set(ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR, false);
+    constitutive_law_options.Set(ConstitutiveLaw::USE_ELEMENT_PROVIDED_STRAIN, true);
+    constitutive_law_options.Set(ConstitutiveLaw::COMPUTE_STRESS, true);
+
+    material_properties.SetValue(YOUNG_MODULUS, 210e9);
+    material_properties.SetValue(POISSON_RATIO, 0.22);
+
+    stress_vector = ZeroVector(6);
+    stress_vector[0] = 5.40984e+06;
+    stress_vector[1] = 5.40984e+06;
+    stress_vector[2] = 1.91803e+07;
+    stress_vector[5] = 1.45804e-10;
+
+    strain_vector = ZeroVector(6);
+    strain_vector[2] = 8.0e-5;
+    strain_vector[5] = 1.6941e-21;
+
+    Matrix deformation_gradient(3, 3);
+    deformation_gradient(0,0) = 1.0;
+    deformation_gradient(1,1) = 1.0;
+    deformation_gradient(2,2) = (1.0 - 8.0e-5);
+    deformation_gradient(2,0) = 0.5 * -1.6941e-21;
+    deformation_gradient(0,2) = 0.5 * -1.6941e-21;
+
+    cl_configuration_values.SetMaterialProperties(material_properties);
+    cl_configuration_values.SetDeformationGradientF(deformation_gradient);
+    cl_configuration_values.SetDeterminantF(MathUtils<double>::DetMat(deformation_gradient));
+    cl_configuration_values.SetStrainVector(strain_vector);
+    cl_configuration_values.SetStressVector(stress_vector);
+    cl_configuration_values.SetOptions(constitutive_law_options);
+    cl_configuration_values.SetProcessInfo(CurrentProcessInfo);
+
+    auto p_constitutive_law = KratosComponents<ConstitutiveLaw>().Get("LinearElastic3DLaw").Clone();
+
+    Matrix C = ZeroMatrix(6, 6);
+    cl_configuration_values.SetConstitutiveMatrix(C);
```

Comment:
Doing this C will be exactly the same as r_tangent_moduli, because we work with references